profiles/management/commands/retire_users.py

```diff
@@ -0,0 +1,129 @@
+`./manage.py retire_users -u foo -u bar -u baz`
+"""
+
+    def create_parser(self, prog_name, subcommand, **kwargs):
+        """
+        create parser to add new line in help text.
+        """
+        parser = super(Command, self).create_parser(prog_name, subcommand)
+        parser.formatter_class = RawTextHelpFormatter
+        return parser
+
+    def add_arguments(self, parser):
+        """create args"""
+        # pylint: disable=expression-not-assigned
+        parser.add_argument(
+            '-u',
+            '--user',
+            action='append',
+            default=[],
+            dest='users',
+            help="Single or multiple user name"
+        ),
+
+    def get_retired_email(self, email):
+        """ Convert user email to retired email format. """
+        return user_util.get_retired_email(email, RETIRED_USER_SALTS, RETIRED_EMAIL_FMT)
+
+    def display_messages(self, message, log_messages, is_error=False):
+        """
+        Display error on console
+        Args:
+            message (str): message to display
+            is_error (bool): is error for styling
+            log_messages (list): Accumulated message
+        """
+        self.stdout.write(message, style_func=self.style.ERROR if is_error else None)
+        log_messages.append(message)
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        users = kwargs.get("users", [])
+        import pdb; pdb.set_trace()
+        if not users:
+            # show error when no user selected.
+            raise CommandError("No user(s) provided. Please provide user(s) using -u or --user.")
+        
+        users = fetch_users(kwargs["users"])
+        for user in users:
+            log_messages = []
+
+            # retire user
+            self.display_messages(f"Retiring user {user}", log_messages)
+
+            if not user.is_active:
+                self.stdout.write(
+                    self.style.ERROR(
+                        "User: '{user}' is already deactivated in MIT BootCamp".format(
+                            user=user
+                        )
+                    )
+                )
+                continue
+
+            # mark user inactive
+            user.is_active = False
+            
+            # Change user password & email
+            email = user.email
+            user.email = self.get_retired_email(user.email)
+            user.set_unusable_password()
+            user.save()
+
+            self.display_messages(f"Email changed from '{email}' to '{user.email}' and password is not useable now", log_messages)
+
+            # reset user social
+            auth_delete_count, _ = UserSocialAuth.objects.filter(user=user).delete()
+            self.display_messages(
+                f"For user {user}: {auth_delete_count} SocialAuth rows deleted",
+                log_messages
+            )
+
+            Interview.objects.filter(applicant=user).update(status='expired')
```

Comment:
@pdpinch , I just found the user association in the Interview model of `Jobma ` application. Either deleting the records would be required here or keeping the interview status to `expired` would be good ?

Reply:
In general I would prefer changing a status instead of deleting a record. But I don't know what the meaning of the statuses are. 

Can you ask @noisecapella or @mbertrand next week?

Reply:
I think setting the status to expired should be adequate (but not sure if it's really necessary).  I'd suggest using a constant in the code if it's going to be changed:

```python
from jobma.contants import EXPIRED
.
.
.

    Interview.objects.filter(applicant=user).update(status=EXPIRED)
```